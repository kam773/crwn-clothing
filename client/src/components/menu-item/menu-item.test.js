import React from 'react';
import MenuItem from './menu-item.component';
import {
  BackgroundImageContainer,
  ContentSubtitle,
  ContentTitle,
  ContentContainer,
} from './menu-item.styles';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

const mockProps = {
  title: 'title',
  imageUrl: 'url',
  size: 'xl',
  linkUrl: 'linkUrl',
  history: {},
  match: {},
};

describe('create snapshot', () => {
  it('captures snapshot of MenuItem component', () => {
    const wrapper = renderer
      .create(
        <MemoryRouter>
          <MenuItem {...mockProps} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('render shallow react component', () => {
  it('renders DUMB component', () => {
    const wrapper = shallow(<MenuItem {...mockProps} />);
    expect(wrapper.length).toEqual(1);
  });
  it('renders BackgroundImageContainer', () => {
    const wrapper = shallow(<MenuItem {...mockProps} />);
    expect(wrapper.find(BackgroundImageContainer)).toBeDefined();
  });
  it('renders ContentTitle', () => {
    const wrapper = shallow(<MenuItem {...mockProps} />);
    expect(wrapper.find(ContentTitle)).toBeDefined();
  });
  it('renders ContentSubtitle', () => {
    const wrapper = shallow(<MenuItem {...mockProps} />);
    expect(wrapper.find(ContentSubtitle)).toBeDefined();
  });
});
