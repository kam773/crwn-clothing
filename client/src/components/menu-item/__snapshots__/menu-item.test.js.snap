// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot captures snapshot of MenuItem component 1`] = `
<div
  className="sc-bdVaJa iAGkBX"
  onClick={[Function]}
  size="xl"
>
  <div
    className="sc-bwzfXH bKoLgt background-image"
  />
  <div
    className="sc-htpNat bxciRy content"
  >
    <span
      className="sc-bxivhb clMuku"
    >
      TITLE
    </span>
    <span
      className="sc-ifAKCX dDPkuj"
    >
      SHOP NOW
    </span>
  </div>
</div>
`;
